planner/core/exhaust_physical_plans.go

```diff
@@ -387,6 +387,101 @@ func (p *LogicalJoin) constructIndexJoin(
 	return []PhysicalPlan{join}
 }
 
+func findIndexScanAndTableScan(p PhysicalPlan) (is *PhysicalIndexScan, ts *PhysicalTableScan) {
+	var tblChildren []PhysicalPlan
+	var idxChildren []PhysicalPlan
+	switch x := p.(type) {
+	case *PhysicalTableReader:
+		tblChildren = x.TablePlans
+	case *PhysicalIndexReader:
+		idxChildren = x.IndexPlans
+	case *PhysicalIndexLookUpReader:
+		tblChildren = x.TablePlans
+		idxChildren = x.IndexPlans
+	default:
+		tblChildren = x.Children()
+	}
+	for _, child := range tblChildren {
+		tmpIS, tmpTS := findIndexScanAndTableScan(child)
+		if tmpIS != nil {
+			is = tmpIS
+		}
+		if tmpTS != nil {
+			ts = tmpTS
+		}
+	}
+	for _, child := range idxChildren {
+		tmpIS, tmpTS := findIndexScanAndTableScan(child)
+		if tmpIS != nil {
+			is = tmpIS
+		}
+		if tmpTS != nil {
+			ts = tmpTS
+		}
+	}
+	if tmpIS, ok := p.(*PhysicalIndexScan); ok {
+		return tmpIS, ts
+	}
+	if tmpTS, ok := p.(*PhysicalTableScan); ok {
+		return is, tmpTS
+	}
+	return is, ts
+}
+
+func (p *LogicalJoin) constructIndexMergeJoin(
+	prop *property.PhysicalProperty,
+	outerIdx int,
+	innerPlan PhysicalPlan,
+	ranges []*ranger.Range,
+	keyOff2IdxOff []int,
+	lens []int,
+	compareFilters *ColWithCmpFuncManager,
+) []PhysicalPlan {
+	indexJoins := p.constructIndexJoin(prop, outerIdx, innerPlan, ranges, keyOff2IdxOff, lens, compareFilters)
+	indexMergeJoins := make([]PhysicalPlan, 0, len(indexJoins))
+	is, ts := findIndexScanAndTableScan(innerPlan)
```

Comment:
Can we directly set `KeepOrder` in this function if we can construct index merge join?